modules/post/windows/gather/credentials/securecrt.rb

```diff
@@ -78,10 +83,10 @@ def enum_session_file(path)
       hostname = Regexp.compile('S:"Hostname"=([^\s]+)').match(file) ? Regexp.last_match(1) : nil
       decrypted_script = Regexp.compile('S:"Login Script V3"=02:([0-9a-f]+)').match(file) ? securecrt_crypto_v2(Regexp.last_match(1)) : nil
       if !decrypted_script.nil?
-        username = decrypted_script.match(/login name:\x1F(\S+)\x1F0\x1Fpass/u)[1]
-        password = decrypted_script.match(/password:\x1F([\S]+)\x1F0/u)[1]
-        domain = decrypted_script.match(/Windows Domain:\x1F([\S]+)\x1F/u) ? decrypted_script.match(/Windows Domain:\x1F([\S]+)\x1F/u)[1] : nil
-        if !domain.nil?
+        username = decrypted_script.match(/login(?: name)?:\x1F(\S+)\x1F(?:[\d])\x1Fpass/u) ? Regexp.last_match(1) : nil
+        password = decrypted_script.match(/password:\x1F([\S]+)\x1F/u) ? Regexp.last_match(1) : nil
+        domain = decrypted_script.match(/Windows Domain:\x1F([\S]+)\x1F/u) ? Regexp.last_match(1) : nil
```

Comment:
We can clean up these ternaries a bit by using the [`String` capture group functionality](https://ruby-doc.org/core-2.7.2/String.html#method-i-5B-5D).
```suggestion
        username = decrypted_script[/login(?: name)?:\x1F(?<login>\S+)\x1F(?:[\d])\x1Fpass/u, 'login']
        password = decrypted_script[/password:\x1F(?<password>[\S]+)\x1F/u, 'password']
        domain = decrypted_script[/Windows Domain:\x1F(?<domain>[\S]+)\x1F/u, 'domain']
```

Reply:
Not going to lie, in my opinion this only serves to make this more confusing than it is already. Personally I think the ternary option looks cleaner and isolates the logic into separate components within the line rather than bunching it all together and intermingling it within the regex like this edit does here.

Otherwise I do think this looks good, just concerned that it confuses the reader more.

Reply:
After further discussion with team members it was decided that we will link to the documentation listed above and add some comments explaining this syntax, but still apply the changes made above, as its useful for people to learn to use this new syntax and it clean up the code in some situations.

Reply:
@gwillcox-r7 I think we'd be good to omit updating the codebase with comments on how the above syntax works. There's pre-existing examples of this syntax within the codebase, and I think if a user isn't aware of a particular Ruby syntax - they'll just need to look up the string api and see that there's an accessor method defined that can take regexes and see how it works

Reply:
@adfoster-r7 Thanks makes sense. Do you think we should also redo the other lines with use `Regex.compile` as well whilst we are at it?

Reply:
@adfoster-r7 Any updates on this?

Reply:
@gwillcox-r7 Sorry for the delay; There shouldn't be any need for Regex.compile as we're not wishing to construct a new regex from a string, the current approach using regex literals should suffice :+1:

Reply:
@adfoster-r7 I'll give this a shot today and see if I can't update the code. Cleaner code is always a win in my books 😄

Reply:
@adfoster-r7 Made some more changes to try fix up both some regex issues and also to fix things to use the String capture group functionality you mentioned to make the code cleaner. Let me know if anything looks odd but hopefully the code should be better and also return better results now.